archive_query_log/parsers/warc_direct_answers.py

```diff
@@ -0,0 +1,272 @@
+from functools import cache
+from itertools import chain
+from typing import Iterable, Iterator
+from urllib.parse import urljoin
+from uuid import uuid5
+
+from click import echo
+from elasticsearch_dsl import Search
+from elasticsearch_dsl.function import RandomScore
+from elasticsearch_dsl.query import FunctionScore, Term, RankFeature, Exists
+# noinspection PyProtectedMember
+from lxml.etree import _Element, tostring  # nosec: B410
+from tqdm.auto import tqdm
+from warc_s3 import WarcS3Store
+
+from archive_query_log.config import Config
+from archive_query_log.namespaces import NAMESPACE_WARC_DIRECT_ANSWER_PARSER, \
+    NAMESPACE_RESULT
+from archive_query_log.orm import Serp, InnerParser, InnerProviderId, \
+    WarcDirectAnswerParserType, WarcDirectAnswerParser, WarcLocation, DirectAnswer, \
+    Result, InnerSerp, DirectAnswerId, InnerDownloader
+from archive_query_log.parsers.warc import open_warc
+from archive_query_log.parsers.xml import parse_xml_tree, safe_xpath
+from archive_query_log.utils.es import safe_iter_scan, update_action
+from archive_query_log.utils.time import utc_now
+
+
+def add_warc_direct_answer_parser(
+        config: Config,
+        provider_id: str | None,
+        url_pattern_regex: str | None,
+        priority: float | None,
+        parser_type: WarcDirectAnswerParserType,
+        xpath: str | None,
+        big_box_xpath: str | None,
+        small_box_xpath: str | None,
+        right_box_xpath: str | None,
```

Comment:
I now think about it and probably it would be best to treat the "big box", "small box", and "right box" as if they were independent direct answers. So for a typical Google page, we could find one or two direct answers (depending on which boxes are available).